app/src/commonMain/kotlin/io/github/cgr/app/io/Unpack.kt

```diff
@@ -5,7 +5,8 @@ import io.github.cgr.app.objects.ConstantLabel
 import io.github.cgr.app.objects.Opcode
 
 fun readProgram(buffer: ByteArray): Program {
-    val unpacker = Unpacker(buffer)
+    require(buffer[0].toInt() == 0xDEAD && buffer[1].toInt() == 0xC0DE) { "Invalid magic number" }
+    val unpacker = Unpacker(buffer.sliceArray(2 until buffer.size))
```

Comment:
`0xDEADCODE` is four bytes, not two.

Reply:
I'm not super familiar with array operations in Kotlin, but in e.g. Swift I might write this as
```swift
precondition(buffer.startsWith([0xDE, 0xAD, 0xC0, 0xDE]), "Invalid magic number")
let unpacker = Unpacker(buffer[4.../* You don't need to specify the upper bound */])
```

Reply:
yes, brain bork